src/ble/CHIPBleServiceData.h

```diff
@@ -64,6 +65,20 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint16_t GetAdvertisementVersion() const
+    {
+        return chip::Encoding::LittleEndian::Get16(DeviceDiscriminator) & kAdvertisementVersionMask;
+    }
+
+    // Use only 4 bits to set advertisement version
+    void SetAdvertisementVersion(uint16_t advertisementVersion)
+    {
+        // Advertisement Version is 4 bit long from 12th to 15th
+        advertisementVersion &= kAdvertisementVersionMask;
+        advertisementVersion |= static_cast<uint16_t>(DeviceDiscriminator[1] << 8u & ~kAdvertisementVersionMask);
+        chip::Encoding::LittleEndian::Put16(DeviceDiscriminator, advertisementVersion);
```

Comment:
Doesn't this always compeletely overwrite `DeviceDiscriminator[0]`?  I don't follow this logic.

Reply:
updated the code